plugins/modules/mail.py

```diff
@@ -348,6 +348,7 @@ def main():
     msg['From'] = formataddr((sender_phrase, sender_addr))
     msg['Date'] = formatdate(localtime=True)
     msg['Subject'] = Header(subject, charset)
+    msg['Message-ID'] = make_msgid(domain='ansible')
```

Comment:
It looks like the `domain` option was only added in Python 3.2, and never backported to Python 2. Let's do the following:
```suggestion
    try:
        msg['Message-ID'] = make_msgid(domain='ansible')
    except TypeError:
        # `domain` is only available in Python 3
        msg['Message-ID'] = make_msgid()
```